pkg/network/route_cache.go

```diff
@@ -187,10 +187,12 @@ type ifEntry struct {
 }
 
 type netlinkRouter struct {
+	sync.Mutex
```

Comment:
nit: prefer to use a field  for sync.Mutex instead of embedding https://github.com/uber-go/guide/blob/master/style.md#embedding-in-structs